tests/functional/instance_manager/test_instances.py

```diff
@@ -0,0 +1,192 @@
+from . import EVENT_1, EVENT_2, EVENT_3
+from . import start_server
+
+"""
+Instance manager tests.
+"""
+
+
+def run_cmd(cmd):
+    print(cmd)
+    proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE)
+
+    out, _ = proc.communicate()
+    print(out)
+    return proc.returncode
+
+
+class Instances(unittest.TestCase):
+    """
+    Server instance manager tests.
+    """
+
+    def setUp(self):
+        # Get the test workspace used to tests.
+        self._test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self._test_workspace)
+
+    def testInitialFile(self):
+        """Tests that initially az instance file is created."""
+
+        self.assertEqual(instance_manager.list(), [],
+                         instance_manager.list())
+        # "The initial instance file was not empty initially.")
+
+    def testServerStart(self):
+        """Started server writes itself to instance list"""
+
+        test_cfg = env.import_test_cfg(self._test_workspace)
+        codechecker_1 = test_cfg['codechecker_1']
+        start_server(codechecker_1, test_cfg, EVENT_1)
+
+        self.assertEqual(len(instance_manager.list()), 1,
+                         "The started server was not appended to the "
+                         "instance list.")
+
+        self.assertEqual(instance_manager.list()[0]['port'],
+                         codechecker_1['viewer_port'],
+                         "The written port in the instance list is not the "
+                         "one the server was started with!")
+
+        self.assertEqual(instance_manager.list()[0]['workspace'],
+                         self._test_workspace,
+                         "Server was brought up with invalid workspace.")
+
+    def testServerStartSecondary(self):
+        """Another started server appends itself to instance list"""
+
+        test_cfg = env.import_test_cfg(self._test_workspace)
+        codechecker_1 = test_cfg['codechecker_1']
+        codechecker_2 = test_cfg['codechecker_2']
+        start_server(codechecker_2, test_cfg, EVENT_2)
+
+        self.assertEqual(len(instance_manager.list()), 2,
+                         "The started server was not appended to the "
+                         "instance list.")
+
+        # Workspaces must match, servers were started in the same workspace.
+        instance_workspaces = [i['workspace'] for i in instance_manager.list()]
+        self.assertEqual(instance_workspaces[0], self._test_workspace,
+                         "Server was brought up with invalid workspace.")
+        self.assertEqual(instance_workspaces[0], instance_workspaces[1],
+                         "Server was brought up with invalid workspace.")
+
+        # Exactly one server should own each port generated
+        instance_ports = [i['port'] for i in instance_manager.list()]
+
+        self.assertTrue(any([True for p
```

Comment:
Isn't there a way to express this in a more concise way? Like` codechecker_1['viewer_port'] in instance_ports` and also the next check.